.github/workflows/main.yml

```diff
@@ -38,14 +38,21 @@ jobs:
 
       - name: Bootstrap poetry
         shell: bash
-        run: |
-          python -m ensurepip
-          python -m pip install --upgrade pip
-          python -m pip install .
+        run: python install-poetry.py -y
```

Comment:
Should we use the script here? I would think the better approach here would be to handle test cases outside of this job. We have gone down the path of using this script before and often times it is has ended up in more issues than solutions.

Reply:
I think we should, yes. It would be weird in my opinion to not use our own documented method of installing Poetry.

Reply:
There are cople of things here.

1. I do not think we should recommend one option over the other. Atleast not when it comes to bootstrap, pip or pipx.
2. I would agree that this helps with our test coverage, however, this has historically been problematic causing us to flap between `pip install` and our bootstrap for various reasons. I would be more agreeable with the solution if the bootstrap mechanism was not a moving target.

All that said, I do not see this as a blocker for this change.

Reply:
Related: https://github.com/python-poetry/poetry-core/pull/144/commits/81fc0d42d806583096392a822202483e85e1b4e8

Reply:
This can move outside this pull request as it should not block progress on this.